kuibit/cactus_grid_functions.py

```diff
@@ -1407,52 +1412,40 @@ def _read_component_as_uniform_grid_data(
         :rtype: :py:class:`~.UniformGridData`
 
         """
-        rx_mesh = re.compile(self._pattern_group_name)
+        # ref_level is an integer like 0, 1, 2, 3, 4, 5. 
+        # So it is prefixed with a "0" to give: 00, 01, 02. 03, 04, 05
+        mesh_name = self.mesh_name+'_lev'+'%02d' % ref_level
```

Comment:
You should use f-strings. It will look something like this
```suggestion
        mesh_name = f"{self.mesh_name}_lev{ref_level:02d}"
```

Reply:
Is `mesh_name = {thorn_name}_{variable_name}`?

Reply:
Mesh names are like:  admbasex_shift_lev02, admbasex_metric_lev00, admbasex_lapse_lev00, weyl_weyl_scalars_lev00
var names are like: admbasex_alp, admbasex_kxx

Not sure what is variable_group_name

Reply:
We do not have thorn name in OpenPMD

Reply:
It should that `ADMBaseX` is the thorn name, and `metric` the group name. This is not reflected at the OpenPMD-file level, but it is a higher-level setting.

Reply:
Also, regular expressions match `rl|lev`, but here you assume that the string is `lev`. What's up with `rl`?